### DISCUSSION TITLE
CONFIG_URL_DRIVER to populated in Android build.gradle file (frontend app)

### DISCUSSION THREAD

**Discussion by kishorgandham** [nammayatri/nammayatri#399] [Q&A]
Hi All, 
I am trying to get the Android version of frontend app running (driver app). I am not sure about the URL to populated for property CONFIG_URL_DRIVER. Can any provide any pointer to what this should be?

I was able to bring the frontend app up (driver version) and was able to get to the login screen and after entering the mobile number, the app is trying to do a POST to CONFIG_URL_DRIVER/auth endpoint. I am not which backend app has the code for handling this /auth call. Any pointers in this regard will help.

**Comment by kishorgandham** [comment]
I was able to resolve the issue by pointing the CONFIG_URL_DRIVER to <dynamic-offer-driver-app>:8016/ui
Is there a default value to be entered for MERCHANT_ID_DRIVER in build.gradle. Second, is there a test mobile number/otp to be used for driver login?

**Reply by vikrantuk** [reply]
@kishorgandham Hey, have you hosted your backend or are you running it on localhost?

**Reply by kishorgandham** [reply]
@vikrantuk I am planning to host the backend, not run on localhost. The issue is lack of clarity on  what some of the fields in build.gradle should be. Without knowing those, it is difficult to make progress on running the Android app.

**Reply by vikrantuk** [reply]
@kishorgandham I'm myself having trouble getting the android app running. Was seeking some help from you to get it at least running.

**Reply by kishorgandham** [reply]
@vikrantuk here is what I did
1. Start local node service by running command `npm run start:android:sandbox`
2. Go to build.gradle file in Android project, in buildTypes {} section, there is a `def host=...` line (line number 68). I modified it to `def host = "192.168.0.225"` 192.168.0.225 is the IP address of the laptop/pc in which the npm server is started in step#1
3. In the same build.gradle file locate `productFlavors` section and assign values to CONFIG_URL_USER, CONFIG_URL_DRIVER, MERCHANT_ID_USER, MERCHANT_ID_DRIVER. I do not know what to assign here and hence I asked this question in this forum. One thing you can try is to point CONFIG_URL_DRIVER to the dynamic offers driver app port

Once the above are done, build the android project and deploy to phone. Assuming the phone and PC are on the same wifi network, you will get the driver UI up, after the nammayatri logo goes away, you will get language selection screen and finally mobile number based OTP.
My view is that the mobile app is talking to the dynamic/static driver app and trying to do auth. We need merchant-id, mobile and OTP that might have been pre-populated in the backend db to complete the auth/login. I have not been able to go past this step.
If you find anything that can help go past this step, do let me know
